dotCMS/src/main/java/com/dotmarketing/business/IdentifierFactoryImpl.java

```diff
@@ -479,51 +467,48 @@ protected Identifier find(final String identifier) throws DotStateException, Dot
 		return check404(id);
 	}
 
-	@Override
-	protected Identifier saveIdentifier(final Identifier id) throws DotDataException {
-		String query;
-		if (id != null) {
-			if (UtilMethods.isSet(id.getId())) {
-
-				if (isIdentifier(id.getId())) {
-                    query = "UPDATE identifier set parent_path=?, asset_name=?, host_inode=?, asset_type=?, syspublish_date=?, sysexpire_date=?, owner=?, create_date=?, asset_subtype=? where id=?";
-				} else{
-					query = "INSERT INTO identifier (parent_path,asset_name,host_inode,asset_type,syspublish_date,sysexpire_date,owner,create_date,asset_subtype,id) values (?,?,?,?,?,?,?,?,?,?)";
-				}
-			} else {
-				id.setId(UUIDGenerator.generateUuid());
-				query = "INSERT INTO identifier (parent_path,asset_name,host_inode,asset_type,syspublish_date,sysexpire_date,owner,create_date,asset_subtype,id) values (?,?,?,?,?,?,?,?,?,?)";
-			}
+    @Override
+    protected Identifier saveIdentifier(final Identifier id) throws DotDataException {
+        if (id == null) {
+            throw new DotRuntimeException("identifier to save cannot be null");
+        }
 
-			DotConnect dc = new DotConnect();
-			dc.setSQL(query);
-
-			dc.addParam(id.getParentPath());
-			dc.addParam(id.getAssetName());
-			dc.addParam(id.getHostId());
-			dc.addParam(id.getAssetType());
-			dc.addParam(id.getSysPublishDate());
-			dc.addParam(id.getSysExpireDate());
-
-            dc.addParam(id.getOwner());
-            dc.addParam(id.getCreateDate());
-            dc.addParam(id.getAssetSubType());
-            dc.addParam(id.getId());
-
-			try{
-				dc.loadResult();
-			}catch(DotDataException e){
-				Logger.error(IdentifierFactoryImpl.class, "saveIdentifier failed:" + e, e);
-				throw new DotDataException(e);
-			}
+        String query;
+        if (UtilMethods.isSet(id.getId())) {
+            if (isIdentifier(id.getId())) {
+                query = "UPDATE identifier set parent_path=?, asset_name=?, host_inode=?, asset_type=?, syspublish_date=?, sysexpire_date=?, owner=?, create_date=?, asset_subtype=? where id=?";
```

Comment:
these scripts (insert/update) can be declared as private constants

Reply:
they could. but this class is All formatted following that pattern I'd rather leave it like so